src/app/rpmostree-ex-builtin-commit2jigdo.c

```diff
@@ -0,0 +1,1167 @@
+                   GCancellable                 *cancellable,
+                   GError                      **error)
+{
+  g_assert_cmpint (*outputdir, ==, '/');
+  g_autofree char *commit = NULL;
+  g_autoptr(GFile) root = NULL;
+  if (!ostree_repo_read_commit (self->repo, provided_commit, &root, &commit, cancellable, error))
+    return FALSE;
+  g_autoptr(GVariant) commit_obj = NULL;
+  if (!ostree_repo_load_commit (self->repo, commit, &commit_obj, NULL, error))
+    return FALSE;
+  g_autoptr(GVariant) commit_inline_meta = g_variant_get_child_value (commit_obj, 0);
+  g_autoptr(GVariant) commit_detached_meta = NULL;
+  if (!ostree_repo_read_commit_detached_metadata (self->repo, commit, &commit_detached_meta,
+                                                  cancellable, error))
+    return FALSE;
+
+  g_print ("Finding reachable objects from target %s...\n", commit);
+  g_autoptr(GHashTable) commit_reachable = NULL;
+  if (!ostree_repo_traverse_commit (self->repo, commit, 0,
+                                    &commit_reachable,
+                                    cancellable, error))
+    return FALSE;
+  GLNX_HASH_TABLE_FOREACH_IT (commit_reachable, it, GVariant *, object,
+                              GVariant *, also_object)
+    {
+      OstreeObjectType objtype;
+      const char *checksum;
+      ostree_object_name_deserialize (object, &checksum, &objtype);
+      if (objtype == OSTREE_OBJECT_TYPE_FILE)
+        g_hash_table_add (self->commit_content_objects, g_strdup (checksum));
+      g_hash_table_iter_remove (&it);
+    }
+  g_print ("%u content objects\n", g_hash_table_size (self->commit_content_objects));
+
+  g_print ("Finding reachable objects from packages...\n");
+  g_autoptr(RpmOstreeRefSack) sack =
+    rpmostree_get_refsack_for_commit (self->repo, commit, cancellable, error);
+  if (!sack)
+    return FALSE;
+
+  hy_autoquery HyQuery hquery = hy_query_create (sack->sack);
+  hy_query_filter (hquery, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
+  g_autoptr(GPtrArray) pkglist = hy_query_run (hquery);
+  g_print ("Building object map from %u packages\n", pkglist->len);
+
+  g_assert_cmpint (pkglist->len, >, 0);
+
+  for (guint i = 0; i < pkglist->len; i++)
+    {
+      DnfPackage *pkg = pkglist->pdata[i];
+      if (!build_objid_map_for_package (self, pkg, cancellable, error))
+        return FALSE;
+    }
+
+  g_print ("%u content objects in packages\n", g_hash_table_size (self->content_object_to_pkg_objid));
+  g_print ("  %u duplicate, %u unused\n",
+           self->n_duplicate_pkg_content_objs, self->n_unused_pkg_content_objs);
+  g_print ("  %u big sizematches, %u/%u nonunique basenames\n",
+           self->duplicate_big_pkgobjects, self->n_nonunique_objid_basenames, self->n_objid_basenames);
+  g_autoptr(GHashTable) new_reachable_big = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+  g_autoptr(GHashTable) new_reachable_small = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
```

Comment:
Minor: can we call these `unpackaged_content_[big,small]` ?

Reply:
The terminology here comes from the static delta code.  I see what you mean though.  I'll add a comment for now: `/* These sets track objects which aren't in the packages */`